From 21c9cc104d706fb5ecd129ea0c6c2d8460ea713d Mon Sep 17 00:00:00 2001
From: Andrew Lunn <andrew@lunn.ch>
Date: Mon, 5 May 2025 19:09:16 -0500
Subject: [PATCH 24/29] net: phy: maxio: Fix return type for phy_read_paged()

All the read functions return an int, so they can be negative error
codes. So using a u32 is wrong.

Signed-off-by: Andrew Lunn <andrew@lunn.ch>
---
 drivers/net/phy/maxio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/maxio.c b/drivers/net/phy/maxio.c
index ffd86c873..4bf07b1a0 100755
--- a/drivers/net/phy/maxio.c
+++ b/drivers/net/phy/maxio.c
@@ -34,7 +34,7 @@ static int maxio_write_page(struct phy_device *phydev, int page)
 
 static int maxio_mae0621a_clk_init(struct phy_device *phydev)
 {
-	u32 workmode, clkmode;
+	int workmode, clkmode;
 	int ret;
 
 	ret = genphy_soft_reset(phydev);
-- 
2.39.5

